rules/rollup.go

```diff
@@ -82,19 +103,58 @@ func (t *rollupTarget) sameTransform(other rollupTarget) bool {
 }
 
 // clone clones a rollup target.
-func (t *rollupTarget) clone() rollupTarget {
+func (t *RollupTarget) clone() RollupTarget {
 	name := make([]byte, len(t.Name))
 	copy(name, t.Name)
 	policies := make([]policy.Policy, len(t.Policies))
 	copy(policies, t.Policies)
-	return rollupTarget{
+	return RollupTarget{
 		Name:     name,
 		Tags:     bytesArrayCopy(t.Tags),
 		Policies: policies,
 	}
 }
 
-func (t rollupTarget) Schema() (*schema.RollupTarget, error) {
+type rollupRuleSnapshotJSON struct {
+	Name         string             `json:"name"`
+	Tombstoned   bool               `json:"tombstoned"`
+	CutoverNanos int64              `json:"cutoverNanos"`
+	TagFilters   map[string]string  `json:"filters"`
+	Targets      []rollupTargetJSON `json:"targets"`
+}
+
+func newRollupRuleSnapshotJSON(rrs rollupRuleSnapshot) rollupRuleSnapshotJSON {
+	targets := make([]rollupTargetJSON, len(rrs.targets))
+	for i, t := range rrs.targets {
+		targets[i] = newRollupTargetJSON(t)
+	}
+
+	return rollupRuleSnapshotJSON{
+		Name:         rrs.name,
+		Tombstoned:   rrs.tombstoned,
+		CutoverNanos: rrs.cutoverNanos,
+		TagFilters:   rrs.rawFilters,
+		Targets:      targets,
+	}
+}
+
+func (rrsj rollupRuleSnapshotJSON) rollupRuleSnapshot() rollupRuleSnapshot {
+	targets := make([]RollupTarget, len(rrsj.Targets))
+	for i, t := range rrsj.Targets {
+		targets[i] = t.rollupTarget()
+	}
+
+	return rollupRuleSnapshot{
```

Comment:
Perhaps better to add a `newRollupRuleSnapshot` ctor to create new `rollupRuleSnapshot` values and reuse it wherever possible?